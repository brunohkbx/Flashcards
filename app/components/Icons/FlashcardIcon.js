import React from 'react';
import PropTypes from 'prop-types';
import { Ionicons } from '@expo/vector-icons';
import { TouchableOpacity } from 'react-native';

const FlashcardIcon = ({ size, color, onPress }) => {
  return (
    <TouchableOpacity onPress={onPress}>
      <Ionicons
        name="ios-card-outline"
        color={color}
        size={size}
      />
    </TouchableOpacity>
  );
};

FlashcardIcon.propTypes = {
  size: PropTypes.number,
  color: PropTypes.string,
  onPress: PropTypes.func
};

export default FlashcardIcon;
