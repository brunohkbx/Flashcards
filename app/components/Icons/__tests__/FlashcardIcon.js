import React from 'react';
import { shallow } from 'enzyme';
import FlashcardIcon from '../FlashcardIcon';

describe('DeleteIcon', () => {
  const setup = propOverrides => {
    const defaultProps = Object.assign({
      onPress: jest.fn()
    }, propOverrides);

    const wrapper = shallow(<FlashcardIcon {...defaultProps} />);

    const press = () => wrapper.simulate('press');

    return { wrapper, press };
  };

  it('renders properly', () => {
    const { wrapper } = setup();

    expect(wrapper).toMatchSnapshot();
  });


  test('onPress calls onPress', () => {
    const mockOnPress = jest.fn();
    const { press } = setup({ onPress: mockOnPress });

    press();

    expect(mockOnPress).toHaveBeenCalled();
  });
});
