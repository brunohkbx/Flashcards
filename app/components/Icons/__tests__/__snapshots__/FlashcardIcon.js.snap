// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteIcon renders properly 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[MockFunction]}
>
  <Icon
    allowFontScaling={false}
    name="ios-card-outline"
    size={12}
  />
</TouchableOpacity>
`;
